Index: oss4-4.2-build2004/setup/elflib.inc
===================================================================
--- oss4-4.2-build2004.orig/setup/elflib.inc	2011-06-14 11:33:23.000000000 -0500
+++ oss4-4.2-build2004/setup/elflib.inc	2011-06-14 11:33:24.000000000 -0500
@@ -81,7 +81,11 @@
     }
   if (hdr->e_ehsize != sizeof (*hdr))
     {
-      fprintf (stderr, "e_ehsize=%x (%x)\n", hdr->e_ehsize, sizeof (*hdr));
+      #if defined(__alpha__) || defined(__s390__) || defined(__ia64__)
+        fprintf (stderr, "e_ehsize=%x (%lx)\n", hdr->e_ehsize, sizeof (*hdr));
+      #else
+        fprintf (stderr, "e_ehsize=%x (%x)\n", hdr->e_ehsize, sizeof (*hdr));
+      #endif
       return 0;
     }
 
